### ISSUE TITLE
Misleading error when incorrect arguments provided to `aks get-credentials`

### ISSUE THREAD

**Issue by yevster** [Azure/azure-cli#7928]
**Describe the bug**

**To Reproduce**
Run 
```bash
az aks get-credentials -g "" -n ""
```

**Expected behavior**
An error message stating that resource group and cluster name values provided are either invalid or not found.

**Actual behavior**
```
The api-version '2018-03-31' is invalid. The supported versions are '2018-09-01,2018-08-01,2018-07-01,2018-06-01,2018-05-01,2018-02-01,2018-01-01,2017-12-01,2017-08-01,2017-06-01,2017-05-10,2017-05-01,2017-03-01,2016-09-01,2016-07-01,2016-06-01,2016-02-01,2015-11-01,2015-01-01,2014-04-01-preview,2014-04-01,2014-01-01,2013-03-01,2014-02-26,2014-04'.
```
This error lead me on a wild goose chase for configuration issues, rather than to re-examine my script that was passing uninitialized variables.

**Environment summary**
apt on Ubuntu / 2.0.51 / 18.04 / bash

**Comment by yevster** [comment]
Looks like a swagger issue, similar to #7256.

**Comment by mboersma** [comment]
I made a similar swagger fix to ACR in Azure/azure-rest-api-specs#4979.

**Comment by palma21** [comment]
Fixed
